glass/algorithm.py

```diff
@@ -245,9 +245,9 @@ def cov_nearest(
         raise ValueError(msg)
 
     # store the normalisation of the matrix
-    norm: NDArray[np.float64] = xp.sqrt(diag)
+    norm = xp.sqrt(diag)
     norm = norm[..., None, :] * norm[..., :, None]
 
     # find nearest correlation matrix
-    corr = cov / xp.where(norm > 0, norm, 1.0)
+    corr = cov / xp.where(norm > 0, norm, xp.asarray([1.0]))
```

Comment:
Why not the scalar argument for `xp.where` here?

Reply:
All the arguments passed to `xp.where` must be of the type `array` - https://data-apis.org/array-api/2023.12/API_specification/generated/array_api.where.html

```py
where(condition: array, x1: array, x2: array, /) → array
```

Reply:
It's fine in the latest API, and circumvents the usual issues with creating new arrays, so I say we should use the scalar.

Reply:
Ah, I didn't see `v2024.12` as JAX and NumPy were still on `v2023.12`. JAX is now on `v2024.12` - https://github.com/jax-ml/jax/commit/a2d62e2d3a332b1d67e0f4ef7a23375182f1646e and NumPy will soon be (next release) - `v2.3.0`.

Reply:
Looks like NumPy is already 2024.22 compatible in the one place `where` we need it to be 🙂